Automattic/jetpack
# RNA Connection: extract connection status into a HOC

#### Changes proposed in this Pull Request:
- connection status fetching extracted into HOC `withConnectionStatus`
- Jetpack Backup and Connection UI are changed to utilized that new HOC
- introducing `@wordpress/data` state for RNA Connection package

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
1. Disconnect Jetpack.
2. Go to Jetpack Backup, the connection screen should appear.
3. Click "Set up Jetpack", complete the connection flow.
4. Confirm that Jetpack Backup is connected.
5. Disconnect Jetpack Backup using the "Disconnect" button. After completing it, you should see the "Set up Jetpack" screen.
6. Go to "Tools -> Connection Manager", complete the connection flow. Confirm that it works well.
